scan/tls_handshake.go

```diff
@@ -30,8 +54,9 @@ func sayHello(host string, ciphers []uint16, vers uint16) (cipherIndex int, err
 	config.MinVersion = vers
 	config.MaxVersion = vers
 	config.CipherSuites = ciphers
+	config.CurvePreferences = curves
 	conn := tls.Client(tcpConn, config)
-	serverCipher, serverVersion, err := conn.SayHello()
+	serverCipher, serverCurveType, serverCurve, serverVersion, err := conn.SayHello()
 	conn.Close()
 	if err != nil {
 		return
```

Comment:
In particular, this would shadow the return of outbound connection errors returned from `conn.readHandshake` at https://github.com/cloudflare/cf-tls/blob/master/tls/conn.go#L777. I'll assume as you say that those aren't interesting, since probably such errors end up getting returned by `tcp.Dial`.